src/ni/vsbuild/packages/Nipkg.groovy

```diff
@@ -113,8 +113,20 @@ class Nipkg extends AbstractPackage {
       def baseVersion = getBaseVersion()
       def fullVersion = getFullVersion()
       def majorVersion = getMajorVersion()
-
-      def additionalReplacements = ['nipkg_version': fullVersion, 'display_version': baseVersion, 'major_version': majorVersion]
+      def minorVersion = getMinorVersion()
+      def updateVersion = getUpdateVersion()
+
+      def additionalReplacements = [
+         'major_version': majorVersion,
+         'minor_version': minorVersion,
+         'update_version': updateVersion,
+         'major_minor_version': "$majorVersion.$minorVersion",
+         'major_minor_update_version': "$majorVersion.$minorVersion.$updateVersion",
+         'major_minor_update_build_version': fullVersion,
+         'nipkg_version': fullVersion,
+         'display_version': baseVersion,
+         'quarterly_display_version': getQuarterlyDisplayVersion(),
```

Comment:
Custom devices will want to use `XB-DisplayVersion: {quarterly_display_version}` instead of `XB-DisplayVersion: {display_version}`; I am leaving the old definition alone to avoid breaking existing builds (both for us and anyone else using these tools).